File: CodeAnalyzerTool/CAT_API/Rule.cs
namespace CAT_API;

public class Rule
{
    public string Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string Category { get; set; }
    public bool IsEnabledByDefault { get; set; }
    public Severity DefaultSeverity { get; set; }

    public Rule(string id, string title, string description, string category, bool isEnabledByDefault, Severity defaultSeverity)
    {
        Id = id;
        Title = title;
        Description = description;
        Category = category;
        IsEnabledByDefault = isEnabledByDefault;
        DefaultSeverity = defaultSeverity;
    }
}

File: CodeAnalyzerTool/CodeAnalyzerService.UI/code-analyzer-service.ui/src/model/Rule.ts
export type Rule = {
    id: number;
    ruleName: string;
    title: string;
    description: string;
    category: string;
    pluginName: string;
    targetLanguage: string;
    isEnabledByDefault: string;
    defaultSeverity: string;
    codeExample: string | null;
    codeExampleFix: string | null;
}